app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -91,50 +92,71 @@ final void setExternalIdService(ExternalIdService externalIdService) {
-            consentHistories.put(subpop.getGuidString(), history);
-        }
-        participant.withConsentHistories(consentHistories);
-        
         ParticipantOptionsLookup lookup = optionsService.getOptions(account.getHealthCode());
-        participant.withSharingScope(lookup.getEnum(SHARING_SCOPE, SharingScope.class));
-        participant.withNotifyByEmail(lookup.getBoolean(EMAIL_NOTIFICATIONS));
-        participant.withExternalId(lookup.getString(EXTERNAL_IDENTIFIER));
-        participant.withDataGroups(lookup.getStringSet(DATA_GROUPS));
-        participant.withLanguages(lookup.getOrderedStringSet(LANGUAGES));
-        participant.withFirstName(account.getFirstName());
-        participant.withLastName(account.getLastName());
-        participant.withEmail(account.getEmail());
-        participant.withStatus(account.getStatus());
-        participant.withCreatedOn(account.getCreatedOn());
-        participant.withRoles(account.getRoles());
-        participant.withId(account.getId());
+        builder.withSharingScope(lookup.getEnum(SHARING_SCOPE, SharingScope.class));
+        builder.withNotifyByEmail(lookup.getBoolean(EMAIL_NOTIFICATIONS));
+        builder.withExternalId(lookup.getString(EXTERNAL_IDENTIFIER));
+        builder.withDataGroups(lookup.getStringSet(DATA_GROUPS));
+        builder.withLanguages(lookup.getOrderedStringSet(LANGUAGES));
+        builder.withFirstName(account.getFirstName());
+        builder.withLastName(account.getLastName());
+        builder.withEmail(account.getEmail());
+        builder.withStatus(account.getStatus());
+        builder.withCreatedOn(account.getCreatedOn());
+        builder.withRoles(account.getRoles());
+        builder.withId(account.getId());
         
         Map<String,String> attributes = Maps.newHashMap();
         for (String attribute : study.getUserProfileAttributes()) {
             String value = account.getAttribute(attribute);
             attributes.put(attribute, value);
         }
-        participant.withAttributes(attributes);
+        builder.withAttributes(attributes);
         
-        if (study.isHealthCodeExportEnabled() && callerRoles.contains(RESEARCHER)) {
-            participant.withHealthCode(account.getHealthCode());
+        if (callerRoles.contains(RESEARCHER)){
+            Map<String,List<UserConsentHistory>> consentHistories = Maps.newHashMap();
+            List<Subpopulation> subpopulations = subpopService.getSubpopulations(study.getStudyIdentifier());
+            for (Subpopulation subpop : subpopulations) {
+                // always returns a list, even if empty
+                List<UserConsentHistory> history = consentService.getUserConsentHistory(study, subpop.getGuid(),
+                        account.getHealthCode(), account.getId());
+                consentHistories.put(subpop.getGuidString(), history);
+            }
+            builder.withConsentHistories(consentHistories);    
```

Comment:
Previously, if a non-researcher called getParticipant to get their own participant info, they would get their own consent history. Now, this is locked to researchers only. Is this an intentional change?

Incidentally, there are no tests in ParticipantService test for a non-researcher, non-developer, non-admin user getting their own participant info.